File: backend/FitnessApp/src/main/java/com/andrea360/fitnessapp/model/TrainingSession.java
package com.andrea360.fitnessapp.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;
import java.time.LocalDateTime;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class TrainingSession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime startTime;

    @Column(nullable = false)
    private LocalDateTime endTime;

    @Positive
    @Column(nullable = false)
    private int maxCapacity;

    @ManyToOne(optional = false)
    @JoinColumn(name = "location_id", nullable = false)
    private Location location;

    @ManyToOne(optional = false)
    @JoinColumn(name = "service_id", nullable = false)
    private TrainingType trainingType;

    @ManyToOne(optional = false)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @OneToMany(mappedBy = "trainingSession")
    private List<Reservation> reservations;

}

File: frontend/src/types/TrainingSession.ts
export interface TrainingSession {
  id: number;
  startTime: string;
  endTime: string;
  maxCapacity: number;
  locationName: string;
  locationAddress: string;
  trainingTypeName: string;
  employeeFirstName: string;
  employeeLastName: string;
  availableSlots: number;
}

export interface CreateTrainingSessionRequest {
  startTime: string;
  endTime: string;
  maxCapacity: number;
  locationId: number;
  trainingTypeId: number;
  employeeId: number;
}